pkg/psmdb/mongos_test.go

```diff
@@ -0,0 +1,310 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mongos-0",
+			Namespace: "default",
+		},
+	}
+
+	tests := map[string]struct {
+		init          func(cl client.Client)
+		cr            *api.PerconaServerMongoDB
+		expectedHost  string
+		expectedError error
+	}{
+		"service not found": {
+			init: func(cl client.Client) {},
+			cr: &api.PerconaServerMongoDB{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+				},
+				Spec: api.PerconaServerMongoDBSpec{
+					ClusterServiceDNSSuffix: "svc.cluster.local",
+					Sharding: api.Sharding{
+						Mongos: &api.MongosSpec{
+							Expose: api.MongosExpose{
+								ServicePerPod: false,
+							},
+						},
+					},
+				},
+			},
+			expectedHost: "",
+		},
+		"clusterip service type": {
+			init: func(cl client.Client) {
+				svc := &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-mongos",
+						Namespace: "default",
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeClusterIP,
+						Ports: []corev1.ServicePort{
+							{
+								Name: "mongos",
+								Port: 27018,
+							},
+							{
+								Name: "random",
+								Port: 12345,
+							},
+						},
+					},
+				}
+				assert.NoError(t, cl.Create(ctx, svc))
+			},
+			cr: &api.PerconaServerMongoDB{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+				},
+				Spec: api.PerconaServerMongoDBSpec{
+					ClusterServiceDNSSuffix: "svc.cluster.local",
+					Sharding: api.Sharding{
+						Mongos: &api.MongosSpec{
+							Expose: api.MongosExpose{
+								ServicePerPod: false,
+							},
+						},
+					},
+				},
+			},
+			expectedHost: "test-cluster-mongos.default.svc.cluster.local:27018",
+		},
+		"err: clusterip service type and port not found": {
+			init: func(cl client.Client) {
+				svc := &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-mongos",
+						Namespace: "default",
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeClusterIP,
+						Ports: []corev1.ServicePort{
+							{
+								Name: "random",
+								Port: 12345,
+							},
+						},
+					},
+				}
+				assert.NoError(t, cl.Create(ctx, svc))
+			},
+			cr: &api.PerconaServerMongoDB{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+				},
+				Spec: api.PerconaServerMongoDBSpec{
+					ClusterServiceDNSSuffix: "svc.cluster.local",
+					Sharding: api.Sharding{
+						Mongos: &api.MongosSpec{
+							Expose: api.MongosExpose{
+								ServicePerPod: false,
+							},
+						},
+					},
+				},
+			},
+			expectedError: errors.New("mongos port not found in service"),
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			s := scheme.Scheme
+			cl := fake.NewClientBuilder().WithScheme(s).Build()
+			assert.NotNil(t, cl)
+			tt.init(cl)
+
+			host, err := MongosHost(ctx, cl, tt.cr, pod, false)
```

Comment:
Given that we are testing `MongosHost` which lives in the `service.go` file, I think we should move this test in ``service_test.go``